sqlglot/dialects/spark.py

```diff
@@ -94,6 +94,10 @@ class Parser(Hive.Parser):
                 unit=exp.var(seq_get(args, 0)),
             ),
             "TRUNC": lambda args: exp.DateTrunc(unit=seq_get(args, 1), this=seq_get(args, 0)),
+            "TIMESTAMP": lambda args: exp.Cast(
+                this=seq_get(args, 0),
+                to="timestamp",
+            ),
```

Comment:
```suggestion
            "TIMESTAMP": lambda args: exp.cast(seq_get(args, 0), "timestamp"),
```

Reply:
Setting `to="timestamp"` is not correct because we need it to be a data type expression.